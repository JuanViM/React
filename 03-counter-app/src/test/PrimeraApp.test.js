import React from 'react';
import PrimeraApp from "../PrimeraApp";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';

describe('Test de mi primer dcomponente PrimeraApp', () => { 
    
    // test('debe de mostrar el mensaje "Hola soy Goku"', () => { 

    //     const saludo = "Hola soy goku";

    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);

    //     expect( getByText(saludo) ).toBeInTheDocument();


    //  })


    test('Debe mostrar <PrimeraApp /> correctamente', () => { 

        const saludo = "hola soy goku";

        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();
       

     });

     test('debe de mostrar el subtitulo enviado por props', () => { 
         
        const saludo = "hola soy goku";

        const wrapper = shallow(
        <PrimeraApp 
        saludo={saludo}
        subtitulo={"Soy un subtitulo"}
        />);

        const textoParrafo = wrapper.find("p").text();
        console.log(textoParrafo);
        expect( textoParrafo).toBe("Soy un subtitulo");
      });

     
 });