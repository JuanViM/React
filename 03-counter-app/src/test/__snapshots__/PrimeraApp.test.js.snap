// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de mi primer dcomponente PrimeraApp Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    hola soy goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Test de mi primer dcomponente PrimeraApp debe de mostrar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
    hola soy goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
